### DISCUSSION TITLE
How to force a number of peaks and position range when fitting?

### DISCUSSION THREAD

**Discussion by nicocopez** [h2020charisma/ramanchada2#85] [Q&A]
Hi,
I am fitting noisy data, but I know that there should be only a peak, and should be in an specific position (since the noise is high, it will alterate the spectrum, but I want to fit only one peak). I know there is only a peak because it is synthetic data.

How can I force that there is only a peak (or two, which it is what I will try next) and that I lies in a specific x axis range?

I am using fit_peak_multimodel, which I know that can be feed some "candidates". However, I can not find how this candidates should be structured. Could you let me know how can I solve this, or where can I look? I was not able to find it in the documentation (most likely because of my searching skills).

Thank you in advance for the help,
nico

**Comment by georgievgeorgi** [comment]
In this discussion there is an example how to create candidates manually: https://github.com/h2020charisma/ramanchada2/discussions/67
I think this is what you need. If you cannot manage to do what you want let me know
